server/src/main/java/org/elasticsearch/index/engine/InternalEngine.java

```diff
@@ -2778,7 +2779,7 @@ private IndexWriterConfig getIndexWriterConfig() {
         iwc.setMaxFullFlushMergeWaitMillis(-1);
         iwc.setSimilarity(engineConfig.getSimilarity());
         iwc.setRAMBufferSizeMB(engineConfig.getIndexingBufferSize().getMbFrac());
-        iwc.setCodec(engineConfig.getCodec());
+        iwc.setCodec(new TrackingPostingsInMemoryBytesCodec(engineConfig.getCodec()));
```

Comment:
I wonder what the overhead is of always wrapping the codec in `TrackingPostingsInMemoryBytesCodec`. Maybe let's quickly run benchmark? (elastic/logs?)

Additionally I wonder whether this should only be done for stateless only.